src/Modules/Legacy/Fields/ExtractSimpleIsosurface.cc

```diff
@@ -70,9 +73,25 @@ void ExtractSimpleIsosurface::execute()
 
     if (isovalueOption && *isovalueOption && !(*isovalueOption)->empty())
     {
-      //TODO: pass entire first column for multiple isovalues
-      double iso = (*isovalueOption)->get(0,0);
-      state->setValue(Parameters::SingleIsoValue, iso);
+      if (state->getValue(Parameters::IsovalueChoice).toString() == "Single")
+      {
+        double iso = (*isovalueOption)->get(0,0);
+        state->setValue(Parameters::SingleIsoValue, iso);
+      }
+      else if (state->getValue(Parameters::IsovalueChoice).toString() == "List")
+      {
+        if (!matrixIs::dense(*isovalueOption)) error("Isovalue input matrix should be dense type");
+        
+        auto mat_iso = castMatrix::toDense (*isovalueOption);
+        double *data = mat_iso->data();
```

Comment:
86-92 replace with:
`std::copy(&mat_iso->col(0), &mat_iso->col(0) + mat_iso->nrows(), std::ostream_iterator<double>(ostr, " "));`

Reply:
This will just get the first column, right? Is there a better way to get all the elements of the matrix?

Reply:
There's only one column there, correct? `DenseMatrixHandle output_matrix( new DenseMatrix(iso_value_vector.size(),1,0.0));`